### ISSUE TITLE
Meal ID not being stored (AJAX Problem)

### ISSUE THREAD

**Issue by louarnos** [ga-wdi-boston/full-stack-project#292]
From curl the post isnt posting the item expected... When i make this post repeatedly it will enter different foods even though they should all be the same.

```
curl --include --request POST http://localhost:3000/meal_items \
  --header "Content-Type: application/json" \
  --data '{
    "meal_item" : {
    "food_for_meal_id" : "4",
    "user_meal_id" : "45"
  }
}'
```

When I make a Post from the front end is doesnt update the meal_items array.... It does from curl though... This is my ajax call:

``` bash
const addFoodToMeal = (success, failure, food_id) => {
  $.ajax({
    method: 'POST',
    url: app.api + 'meal_items/',
    data: {
      "meal_item" : {
        "food_for_meal_id" : food_id,
        "user_meal_id" : currentMeal.local.id
      },
    }
  })
  .done(success)
  .fail(failure);
};
```

I have no idea whats going on.

**Comment by laurenfazah** [comment]
Is it hitting the rails server? Are there errors from your rails console?

**Comment by laurenfazah** [comment]
When it does post with different foods, what is your rails console logging?

**Comment by louarnos** [comment]
It is hitting the rails server. When I use a curl request it gives me:

```
Started POST "/meal_items" for ::1 at 2016-05-03 13:40:17 -0400
Processing by MealItemsController#create as */*
  Parameters: {"meal_item"=>{"food_for_meal_id"=>"4", "user_meal_id"=>"45"}}
   (0.2ms)  BEGIN
  SQL (0.5ms)  INSERT INTO "meal_items" ("food_for_meal_id", "user_meal_id", "created_at", "updated_at") VALUES ($1, $2, $3, $4) RETURNING "id"  [["food_for_meal_id", 4], ["user_meal_id", 45], ["created_at", "2016-05-03 17:40:17.893531"], ["updated_at", "2016-05-03 17:40:17.893531"]]
   (15.0ms)  COMMIT
[active_model_serializers]   Food Load (0.2ms)  SELECT  "foods".* FROM "foods" WHERE "foods"."id" = $1 LIMIT 1  [["id", 4]]
[active_model_serializers]   Meal Load (0.2ms)  SELECT  "meals".* FROM "meals" WHERE "meals"."id" = $1 LIMIT 1  [["id", 45]]
[active_model_serializers]   MealItem Load (0.2ms)  SELECT "meal_items".* FROM "meal_items" WHERE "meal_items"."user_meal_id" = $1  [["user_meal_id", 45]]
[active_model_serializers]   Food Load (0.4ms)  SELECT "foods".* FROM "foods" INNER JOIN "meal_items" ON "foods"."id" = "meal_items"."food_for_meal_id" WHERE "meal_items"."user_meal_id" = $1  [["user_meal_id", 45]]
[active_model_serializers] Rendered MealItemSerializer with ActiveModel::Serializer::Adapter::Json (15.83ms)
Completed 201 Created in 35ms (Views: 15.5ms | ActiveRecord: 16.7ms)
```

When I use the ajax call it gives me:

```
Started POST "/meal_items/" for ::1 at 2016-05-03 13:41:31 -0400
Processing by MealItemsController#create as */*
  Parameters: {"meal_item"=>{"food_for_meal_id"=>"32"}}
   (0.2ms)  BEGIN
  SQL (0.4ms)  INSERT INTO "meal_items" ("food_for_meal_id", "created_at", "updated_at") VALUES ($1, $2, $3) RETURNING "id"  [["food_for_meal_id", 32], ["created_at", "2016-05-03 17:41:31.692840"], ["updated_at", "2016-05-03 17:41:31.692840"]]
   (0.5ms)  COMMIT
[active_model_serializers]   Food Load (0.4ms)  SELECT  "foods".* FROM "foods" WHERE "foods"."id" = $1 LIMIT 1  [["id", 32]]
[active_model_serializers] Rendered MealItemSerializer with ActiveModel::Serializer::Adapter::Json (2.91ms)
Completed 201 Created in 8ms (Views: 3.5ms | ActiveRecord: 1.4ms)
```

**Comment by louarnos** [comment]
This first response is when it would post with different foods, despite the ids being the same in every curl request

**Comment by laurenfazah** [comment]
Is `currentMeal.local.id` defined?  Have you modified your serializers recently?

**Comment by louarnos** [comment]
yes @RDegnen and I modified the serializers. currentMeal.local.id is set when I either create or get meal. I save the object locally to use it again later.

**Comment by louarnos** [comment]
This is my meal_item serializer:

```
class MealItemSerializer < ActiveModel::Serializer
  attributes :id, :serv_qty
  has_one :food_for_meal
  has_one :user_meal
end
```

**Comment by RealWeeks** [comment]
put a debugger right after you open addFoodToMeal and post the value of the currentMeal object and all objects it contains

**Comment by louarnos** [comment]
my DOM can't see currentMeal... When I create a meal I set it locally, and print it to the console. It says:

``` js
Object {meal: Object}
meal
:
Object
created_at
:
"2016-05-03T18:15:50.211Z"
food_for_meals
:
Array[0]
id
:
69
meal_items
:
Array[0]
meal_type
:
"stuff"
user
:
Object
```

**Comment by louarnos** [comment]
Ya its undefined. I'm not sure why.

**Comment by RealWeeks** [comment]
Does this help you on the path to solving your issue or do you need more guidance?

**Comment by louarnos** [comment]
To a degree. I dont understand why its not defined.

**Comment by louarnos** [comment]
I console.log and make use of the data before this call.

**Comment by GA-MEB** [comment]
@louarnos 
Try replacing the `success` handler function in your AJAX with the following:

``` js
(data)=>{ debugger; }
```

What do you see?

**Comment by louarnos** [comment]
for currentMeal?

**Comment by GA-MEB** [comment]
For the value of `data` - the data that you're getting back from the POST request.

**Comment by louarnos** [comment]
its a meal_item object:

```
Object {meal: Object}
meal
:
Object
created_at
:
"2016-05-03T18:47:44.280Z"
food_for_meals
:
Array[0]
id
:
74
meal_items
:
Array[0]
meal_type
:
"233"
user
:
Object
__proto__
:
Object
__proto__
:
Object
```

**Comment by GA-MEB** [comment]
Assigning myself.

**Comment by louarnos** [comment]
I was not actually passing meal_id successfully to my ajax call. Attempting to pass meal_id as an argument led to me realize that I had to redefine how I was exporting currentMeal. Essentially my nesting was too extensive and was able to be shortened by doing:

```
const local = {}

module.exports = local;

```

as opposed to:

```
const local = {}

module.exports = {
local
};
```

Once I understood the nesting better through troubleshooting in the DOM, I was able to successfully pass currentMeal to the ajax call.
